StackViewController/AutoScrollView.swift

```diff
@@ -28,34 +28,121 @@ open class AutoScrollView: UIScrollView {
         }
         didSet {
             if let contentView = contentView {
+                contentView.translatesAutoresizingMaskIntoConstraints = false
                 addSubview(contentView)
                 updateContentViewConstraints()
             }
         }
     }
     fileprivate var contentViewConstraints: [NSLayoutConstraint]?
     
+    /// This layout guide follows the `contentView` and honors the `horizontallyCompactContentWidth`
+    /// and `horizontallyRegularContentWidth` values.
+    public var contentViewLayoutGuide = UILayoutGuide()
+    private var contentViewLayoutGuideConstraints: [NSLayoutConstraint] = []
+    
+    /// This setting determines how the content should be laid out in a horizontally compact environment.
+    public var horizontallyCompactContentWidth: ContentWidth = .matchScrollViewWidth {
+        didSet {
+            guard horizontallyCompactContentWidth != oldValue else { return }
+            
+            switch traitCollection.horizontalSizeClass {
+            case .compact: updateContentViewConstraints()
+            case .regular, .unspecified: break // no-op
+            @unknown default: break
+            }
+        }
+    }
+    
+    /// This setting determines how the content should be laid out in a horizontally regular environment.
+    public var horizontallyRegularContentWidth: ContentWidth = .matchScrollViewWidth {
+        didSet {
+            guard horizontallyRegularContentWidth != oldValue else { return }
+            
+            switch traitCollection.horizontalSizeClass {
+            case .regular: updateContentViewConstraints()
+            case .compact, .unspecified: break // no-op
+            @unknown default: break
+            }
+        }
+    }
+    
     override open var contentInset: UIEdgeInsets {
```

Comment:
All that being said, I can see how different insets for each size class would be beneficial. We don't need it and as far as I know there haven't been any requests for it though. 

Since you gave the approval, I'm going to assume it's good to merge as is. If you'd like me to work on any changes to the `contentInset`, let me know and I can work on it in a separate PR.